recipes/wt/all/test_package/CMakeLists.txt

```diff
@@ -5,11 +5,11 @@ include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
 conan_basic_setup(TARGETS)
 
 add_executable(${PROJECT_NAME} test_package.cpp)
-set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 11)
```

Comment:
Can the C++ standard be controlled via Conan here instead of forcing it in the CMakeLists.txt?

Reply:
@jwillikers 
Sorry, I don't know the way to force C++ standard by conan.

Reply:
Can you use [conan.tools.build.check_min_cppstd](https://docs.conan.io/en/latest/reference/conanfile/tools/build.html#conan-tools-build-check-min-cppstd) in the `validate()` method of the package?

Reply:
As I know, `conan.tools.build.check_min_cppstd` only checks cppstd settings and doesn't set C++ standard options in CXXFLAGS
Is there undocumented side effect in `conan.tools.build.check_min_cppstd`?

Reply:
What requires `C++11`, the `test_package.cpp` file or `wt`?

Reply:
And does the old CMake generator not set it? I know that `CMakeToolchain` sets the C++ standard appropriately, but I I thought the `cmake` generator did, too.

Reply:
@jwillikers 
Now I agree with you.
I am going to switch CMakeToolChain when I add a version to an existing recipe or add a new recipe.

But this PR is intended to update dependencies.
So I want to continue using CMake generator.

Reply:
Keep that property, do not remove it! The way to manage the C++ standard with CMake helper is declaring `settings.compiler.cppstd` which is not configured and `None` by default. More information: https://docs.conan.io/en/latest/reference/build_helpers/cmake.html#definitions

Without that `set_property` it will be used the default version supported by the compiler.

The `cppstd` was introduce in middle of 1.x, thus, it was not possible to use it as standard feature, and that's why your default Conan profile does not have it declared.

Reply:
Sounds good. Sorry about the tangent here.

Reply:
@uilianries Should the C++ standard be set via the CMake cache variable `CMAKE_CXX_STANDARD`, which allows being overridden or should it set the target property or use a non-cache variable so that it can't be overriden?

Reply:
@jwillikers Use Cmake Target always, it's a good practice.